docs/developer-guide/etendo-classic/how-to-guides/how-to-create-a-hql-based-table.md

```diff
@@ -0,0 +1,171 @@
+  * Include the string _@additional_filters@_ inside a where clause in the HQL Query. That string will be replaced with the mandatory client and organization filters, and with the filters generated from the criteria included in the datasource request.
+  * All HQL queries should include a column that acts as its primery key. The alias of this column should be _id_.
+  * If a HQL query contains several from clauses because it uses subqueries in the select clause or in the where clause, then the MAINFROM text must be used instead of FROM in the main from clause.
+
+![](../../../assets/developer-guide/etendo-classic/how-to-guides/How_to_create_a_HQL_Based_Table-1.png)
+
+### Column Definition
+
+The _Create Columns from DB_ process does not apply to HQL based tables, so they have to be created manually.
+
+Considerations in the definition of columns of HQL tables in the application dictionary:
+
+  * The _DB Column Name_ and the _Name_ of the column must be the alias used for the column in the HQL Query.
+  * The _Clause Left Part_ field must be filled with the left part definition of the column in the HQL query. For instance, if a column is defined in the HQL query is `e.name as countryName` , then the _Name_ and _DB Column Name_ of the column must be `countryName` and its _Clause Left Part_ must be `e.name`.
+
+To include a foreign key in a column of a HQL table, the column must point to the referenced entity itself, not to its id. Then, when that column is defined in the application dictionary, the _Table_ reference must be used, and the proper table instance must be selected in the _Reference Search Key_ field.
+
+For instance, if `e` is the alias of the Country table and we want to include its currency, we must enter `e.currency as countryCurrency` in the HQL query. Then we must select the _Table_ reference in the definition of that column in the application dictionary and `c_currency` in the _Reference Search Key_ field.
+
+## Tab and Fields Definition
+
+The tab and the fields of a HQL based table are defined in the application dictionary in the same way as with standard tables.
+
+The only restriction is that all tabs associated with HQL based tables will be read only.
+
+## Advanced Topics
+
+### HQL Inserters
+
+HQL Inserters allow to add code to a HQL query at a point specified in the HQL Query defined in the application dictionary.
+
+#### HQL Insertion points in the HQL Query
+
+To define a HQL Insertion point in your HQL query you need to use the `@insertion_point#@` string, `#` being the index of the insertion point. The
+index of the first injection point must be 0, the index for the second must be 1, etc. There cannot be gaps in the indexes of insertion points.
+
+Insertion points can only be defined in where clauses. If the HQL Query defines an insertion point but then there are no `HQLInserter` defined for it, then the insertion point will be replaced by _1 = 1_ in the HQL query.
```

Comment:
Highlight where clauses based on your preference:
Insertion points can only be defined in  `where clauses`